packages/react-components/react-charts-preview/library/src/utilities/utilities.ts

```diff
@@ -0,0 +1,1701 @@
+  if (xAxisElement) {
+    d3Select(xAxisElement).call(xAxis).selectAll('text').attr('aria-hidden', 'true');
+  }
+  const tickValues = ((tickParams.tickValues as Date[] | undefined) ?? xAxisScale.ticks(tickCount)).map(
+    xAxis.tickFormat()!,
+  );
+  return { xScale: xAxisScale, tickValues };
+}
+
+/**
+ * Create String X axis
+ * Currently using for only Vetical stacked bar chart and grouped vertical bar chart
+ * @export
+ * @param {IXAxisParams} xAxisParams
+ * @param {ITickParams} tickParams
+ * @param {string[]} dataset
+ * @returns
+ */
+export function createStringXAxis(
+  xAxisParams: IXAxisParams,
+  tickParams: ITickParams,
+  dataset: string[],
+  culture?: string,
+) {
+  const {
+    domainNRangeValues,
+    xAxistickSize = 6,
+    tickPadding = 10,
+    xAxisPadding = 0.1,
+    xAxisInnerPadding,
+    xAxisOuterPadding,
+    containerWidth,
+    hideTickOverlap,
+  } = xAxisParams;
+  const xAxisScale = d3ScaleBand()
+    .domain(dataset!)
+    .range([domainNRangeValues.rStartValue, domainNRangeValues.rEndValue])
+    .paddingInner(typeof xAxisInnerPadding !== 'undefined' ? xAxisInnerPadding : xAxisPadding)
+    .paddingOuter(typeof xAxisOuterPadding !== 'undefined' ? xAxisOuterPadding : xAxisPadding);
+
+  let tickValues = (tickParams.tickValues as string[] | undefined) ?? dataset;
+  const tickFormat = (domainValue: string, _index: number) => {
+    return convertToLocaleString(domainValue, culture) as string;
+  };
+  if (hideTickOverlap) {
+    let nonOverlappingTickValues = [];
+    const tickSizes = tickValues.map((value, index) =>
+      calculateLongestLabelWidth([tickFormat(value, index)], '.fui-cart__xAxis text'),
+    );
+    // for LTR
+    let start = 0;
+    let end = containerWidth;
+    let sign = 1;
+    const range = xAxisScale.range();
+    if (range[1] - range[0] < 0) {
+      // for RTL
+      start = containerWidth;
+      end = 0;
+      sign = -1;
+    }
+    for (let i = tickValues.length - 1; i >= 0; i--) {
+      const tickPosition = xAxisScale(tickValues[i])!;
+      if (
+        sign * (tickPosition - (sign * tickSizes[i]) / 2 - start) >= 0 &&
+        sign * (tickPosition + (sign * tickSizes[i]) / 2 - end) <= 0
+      ) {
+        nonOverlappingTickValues.push(tickValues[i]);
+        end = tickPosition - sign * (tickSizes[i] / 2 + 10);
+      }
+    }
+    nonOverlappingTickValues = nonOverlappingTickValues.reverse();
+    tickValues = nonOverlappingTickValues;
+  }
+
+  const xAxis = d3AxisBottom(xAxisScale)
+    .tickSize(xAxistickSize)
+    .tickPadding(tickPadding)
+    .tickValues(tickValues)
+    .tickFormat(tickFormat);
+
+  if (xAxisParams.xAxisElement) {
+    d3Select(xAxisParams.xAxisElement).call(xAxis).selectAll('text').attr('aria-hidden', 'true');
+  }
+  return { xScale: xAxisScale, tickValues: tickValues.map(xAxis.tickFormat()!) };
+}
+
+export function isRtl() {
+  const { dir } = useFluent(); // "dir" returns "ltr" or "rtl"
+  return dir === 'rtl';
+}
```

Comment:
this is breaking the rules of hooks:

```suggestion
export function useRtl() {
  const { dir } = useFluent(); // "dir" returns "ltr" or "rtl"
  return dir === 'rtl';
}
```